### PR TITLE
adding feed page

### PR DESCRIPTION
**Pull request by marinaRomanchuk**

### CODE REVIEW

File: src/components/Feed.js [hunk 1]
```diff
@@ -1,9 +1,234 @@
...
+}
+
+async function getListOfComments(AuthStr, post_id) {
+    return axios.get("http://" + API_DOMAIN + "/api/comments/", { headers: { Authorization: AuthStr }, params: { post_id: post_id,}
+    })
+        .then(result => result.data);
+}
+
+async function getUser(AuthStr, user_id) {
+    const user_str = ''.concat(user_id);
```
Comment (revimi) [inline src/components/Feed.js]: In JS there is a standard to use camelCase
for example, `userId`

File: src/components/Feed.js [hunk 2]
```diff
@@ -1,9 +1,234 @@
...
+import dislike_empty  from '../dislike_empty.png';
+import comment_empty  from '../comment.png';
+import comment_full  from '../comment_full.png';
 
-const Feed = () => (
-    <div className="container feed">
-    </div>
-);
+
+async function getListOfPosts(AuthStr) {
```
Comment (revimi) [inline src/components/Feed.js]: Please start var name with lowercase
```suggestion
async function getListOfPosts(authStr) {
```

File: src/components/Feed.js [hunk 3]
```diff
@@ -1,9 +1,234 @@
...
+    useEffect(() => {
+        const handleSome = async () => {
+            setLoading(true);
+            const posts = await getListOfPosts(AuthStr);
+
+            let ind = 0;
+            for (let item of posts) {
+                const likes = await getLikesCount(AuthStr, item.id);
+                const comments = await getListOfComments(AuthStr, item.id);
+                const user = await getUser(AuthStr, item.user_id);
```
Comment (revimi) [inline src/components/Feed.js]: We can probably update the backend app to return the user object attached to a post object

Comment (revimi) [reply src/components/Feed.js]: As well as the likes and comments count

Comment (revimi) [reply src/components/Feed.js]: And make request to get the comments list just before we show them, not request them in advance

File: src/components/Feed.js [hunk 4]
```diff
@@ -1,9 +1,234 @@
...
+            setLoading(false);
+        }
+        handleSome();
+    }, []);
+
+    const clickLike = (post_id, ind) => {
+        const newPostList = postList.postList;
+
+        if (newPostList[ind].has_liked) {
+            deleteLike(AuthStr, post_id, 1);
```
Comment (revimi) [inline src/components/Feed.js]: why do we have int instead of bool?

File: src/components/Feed.js [hunk 5]
```diff
@@ -1,9 +1,234 @@
...
+                posts[ind].show_comments = false;
+                ind = ind + 1;
+            }
+            setPostList({postList: posts});
+            setLoading(false);
+        }
+        handleSome();
+    }, []);
+
+    const clickLike = (post_id, ind) => {
```
Comment (revimi) [inline src/components/Feed.js]: What `ind` stands for? Index? Let's have more readable variables names

File: src/components/Feed.js [hunk 6]
```diff
@@ -1,9 +1,234 @@
...
+    const [ loading, setLoading ] = useState(true);
+    const AuthStr = 'Token '.concat(token);
+
+    useEffect(() => {
+        const handleSome = async () => {
+            setLoading(true);
+            const posts = await getListOfPosts(AuthStr);
+
+            let ind = 0;
+            for (let item of posts) {
```
Comment (revimi) [inline src/components/Feed.js]: What is `item`? I can't get it from the first look. Readable vars names will save a lot of time maintaining the application

**Review by revimi — COMMENTED** [review]
Looks good. First review is done, will take another look after you address the comments

File: src/components/Feed.js (hunk 1 above)
Comment (marinaRomanchuk) [reply src/components/Feed.js]: Variable names changed

File: src/components/Feed.js (hunk 2 above)
Comment (marinaRomanchuk) [reply src/components/Feed.js]: Done

File: src/components/Feed.js (hunk 3 above)
Comment (marinaRomanchuk) [reply src/components/Feed.js]: Done for user information and comments

File: src/components/Feed.js (hunk 4 above)
Comment (marinaRomanchuk) [reply src/components/Feed.js]: Changed to bool

File: src/components/Feed.js (hunk 5 above)
Comment (marinaRomanchuk) [reply src/components/Feed.js]: Done

File: src/components/Feed.js [hunk 7]
```diff
@@ -1,9 +1,241 @@
...
+    useEffect(() => {
+        const handleSome = async () => {
+            setLoading(true);
+            const posts = await getListOfPosts(authStr);
+
+            let index = 0;
+            for (let post of posts) {
+                const likes = await getLikesCount(authStr, post.id);
+
+                posts[index].likesCount = likes.likes_count;
```
Comment (revimi) [inline src/components/Feed.js]: Doesn't it work this way?
```suggestion
                post.likesCount = likes.likes_count;
```

Comment (revimi) [reply src/components/Feed.js]: Though it's better to use another approach on the backend and return posts together with their stats

**Review by revimi — APPROVED** [review]

File: src/components/Feed.js (hunk 7 above)
Comment (marinaRomanchuk) [reply src/components/Feed.js]: Done

**Review by revimi — APPROVED** [review]

**Pull request merged**